File: Seminar1/java/VendingMachine/src/Domain/HotDrink.java
package Domain;

public class HotDrink extends Product {

    /**
     * Класс Горячих Напитков - наследник класса Product.
     * Содержит дополнительное целочисленное поле temperature, для указания
     * температуры продукта.
     * Поле введено в конструктор класса, для него определён getter и setter,
     * переопределён метод toString.
     */

    private int temperature;

    public HotDrink(int price, int place, String name, long id, int temp) {
        super(price, place, name, id);
        this.temperature = temp;
    }

    public int getTemperature() {
        return temperature;
    }

    public void setTemperature(int temperature) {
        this.temperature = temperature;
    }

    @Override
    public String toString() {
        return super.toString() + "temperature=" + temperature + "\n";
    }
}

File: Seminar1/python/VendingMachine/Domain/hot_drink.py
from Domain.product import Product

class Hot_Drink(Product):
    def __init__(self, price, place, name, id):
        super().__init__(price, place, name, id)
        self.temp = 70

    def __init__(self, price, place, name, id, temp):
        super().__init__(price, place, name, id)
        self.temp = temp

    def getBottleVolume(self):
        return self.temp

    def setBottleVolume(self, temp):
        self.temp = temp

    def __str__(self):
        return super().__str__() + "\ntemperature=" + str(self.temp)